lib/std/os/windows.zig

```diff
@@ -980,31 +1029,64 @@ pub fn GetFinalPathNameByHandle(
     fmt: GetFinalPathNameByHandleFormat,
     out_buffer: []u16,
 ) GetFinalPathNameByHandleError![]u16 {
-    // Get normalized path; doesn't include volume name though.
-    var path_buffer: [@sizeOf(FILE_NAME_INFORMATION) + PATH_MAX_WIDE * 2]u8 align(@alignOf(FILE_NAME_INFORMATION)) = undefined;
-    try QueryInformationFile(hFile, .FileNormalizedNameInformation, path_buffer[0..]);
-
-    // Get NT volume name.
+    var path_buffer: [std.math.max(@sizeOf(FILE_NAME_INFORMATION), @sizeOf(OBJECT_NAME_INFORMATION)) + PATH_MAX_WIDE * 2]u8 align(@alignOf(FILE_NAME_INFORMATION)) = undefined;
     var volume_buffer: [@sizeOf(FILE_NAME_INFORMATION) + MAX_PATH]u8 align(@alignOf(FILE_NAME_INFORMATION)) = undefined; // MAX_PATH bytes should be enough since it's Windows-defined name
-    try QueryInformationFile(hFile, .FileVolumeNameInformation, volume_buffer[0..]);
 
-    const file_name = @ptrCast(*const FILE_NAME_INFORMATION, &path_buffer[0]);
-    const file_name_u16 = @ptrCast([*]const u16, &file_name.FileName[0])[0 .. file_name.FileNameLength / 2];
+    var file_name_u16: []const u16 = undefined;
+    var volume_name_u16: []const u16 = undefined;
+    if ((comptime (targetVersionIsAtLeast(WindowsVersion.win10_rs4) != true)) //need explicit comptime, because error returns affect return type
+        and !runtimeVersionIsAtLeast(WindowsVersion.win10_rs4))
```

Comment:
I was actually wondering whether we should cache that version in a global (that is, module-scope) variable - not just for this code path, but for all of `std.os.windows`, but it seemed like too big of a change to just throw in unsupervised.
So would an atomic version number `u32` in `std.zig.system.windows` be fitting, initialized once and then used instead of the `RtlGetVersion` call?
Or is there some benefit to having this particular flag be in its own variable directly?

Reply:
Personally, I'm not sure I'm a big fan of having a global/thread-local state lying around, but on the other hand, calling `RtlGetVersion` every single time we execute `GetFinalPathNameByHandle` is a performance hog indeed. @andrewrk what are your thoughts on this?

Reply:
@kubkon Right, that's why I originally refrained.
Maybe this is also a good place to mention that (A) we can always incrementally improve this in subsequent commits, and (B) `NtQueryObject` works for ALL windows versions (for now).
So if just removed the other path (or are satisfied with only the exhaustive `comptime` check), the runtime check instantly becomes superfluous.

Reply:
`RtlGetVersion` should be pretty cheap: its a read from the PEB.
We could read from the PEB ourselves too, but there shouldn't be much of a difference.